src/handlers/user.rs

```diff
@@ -82,11 +87,18 @@ pub async fn get_user_without_auth(
     Ok(Json(user))
 }
 
+const MAX_BIO_LENGTH: usize = 10;
+
 pub async fn update_user_bio(
     Extension(auth_data): Extension<AuthData>,
     State(state): State<Arc<AppState>>,
     Json(bio): Json<Bio>,
 ) -> Result<Json<User>, AppError> {
+    if bio.len() > MAX_BIO_LENGTH {
+        return Err(AppError::BioTooLong(
+            "Bio exceeds maximum length".to_string(),
+        ));
+    }
```

Comment:
```suggestion
    if bio.bio.len() > MAX_BIO_LENGTH {
        return Err(AppError::BioTooLong(
            "Bio exceeds maximum length".to_string(),
        ));
    }
```
You can directly access the struct variable to avoid implementing a `.len()` function. This should also fix your clippy warning. It's perfectly fine as is, but you should always aim to code less unless you are trying to make the code more readable or you absolutely have to.

Reply:
i know that bio.bio is a bit weird looking. We can change the struct field name to `inner` and use `#[serde(rename = "bio")]`

Reply:
While thinking about this, if you move the whole length check inside the struct implementation, then it might be worth it to keep using it.